docs/take-to-prod.md

```diff
@@ -0,0 +1,85 @@
+}
+```
+
+## How to deploy your application when you have multiple dependencies
+### Scenarios
+#### 1. SparkSession code in one project that references another project including the business logic
+This would be the use case when you have `SparkSession` code in one project (e.g. mySparkApp.csproj) and business logic (UDFs) in another project (e.g. businessLogic.csproj).
+#### 2. SparkSession code references a function from a Nuget package that has been installed in the csproj
+This would be the use case when `SparkSession` code references a function from a Nuget package in the same project (e.g. mySparkApp.csproj).
+#### 3. SparkSession code references a function from a DLL on the user's machine
+This would be the use case when `SparkSession` code reference business logic (UDFs) on the user's machine (e.g. `SparkSession` code in the mySparkApp.csproj and businessLogic.dll on a different machine). 
+#### 4. SparkSession code references functions and business logic from multiple projects/solutions that themselves depend on multiple Nuget packages
+This would be a more complex use case when you have `SparkSession` code reference business logic (UDFs) and functions from Nuget packages in multiple projects and/or solutions.
+
+### Package your application
+Please see detailed steps [here](https://github.com/dotnet/spark/tree/master/deployment#preparing-your-spark-net-app) on how to build, publish and zip your application. After packaging your .NET for Apache Spark application, you will have a zip file (e.g. mySparkApp.zip) which has all the dependencies.
+
+### Launch your application
+#### 1. Using spark-submit
+```shell
+spark-submit \
+--class org.apache.spark.deploy.dotnet.DotnetRunner \
+--master yarn \
+--deploy-mode cluster \
+--conf spark.yarn.appMasterEnv.DOTNET_ASSEMBLY_SEARCH_PATHS=./udfs \
+--conf spark.yarn.appMasterEnv.DOTNET_ASSEMBLY_SEARCH_PATHS=./myLibraries.zip \
+--archives hdfs://<path to your files>/businessLogics.zip#udfs,hdfs://<path to your files>/myLibraries.zip \
+hdfs://<path to jar file>/microsoft-spark-<spark_majorversion.spark_minorversion.x>-<spark_dotnet_version>.jar \
+hdfs://<path to your files>/mySparkApp.zip mySparkApp <app arg 1> <app arg 2> ... <app arg n>
+```
+#### 2. Using Apache Livy
+```shell
+{
+    "file": "adl://<cluster name>.azuredatalakestore.net/<some dir>/microsoft-spark-<spark_majorversion.spark_minorversion.x>-<spark_dotnet_version>.jar",
+    "className": "org.apache.spark.deploy.dotnet.DotnetRunner",
+    "conf": {"spark.yarn.appMasterEnv.DOTNET_ASSEMBLY_SEARCH_PATHS": "./udfs, ./myLibraries.zip"},
+    "archives": ["adl://<cluster name>.azuredatalakestore.net/<some dir>/businessLogics.zip#udfs”, "adl://<cluster name>.azuredatalakestore.net/<some dir>/myLibraries.zip”],
+    "args": ["adl://<cluster name>.azuredatalakestore.net/<some dir>/mySparkApp.zip","mySparkApp","<app arg 1>","<app arg 2>,"...","<app arg n>"]
+}
+```
```

Comment:
You should say somewhere that scenarios 1 and 2 can also be submitted using these submission examples.

Reply:
Do you mean scenarios 1 and 2 in single dependency session?
They can submit it using the example here, but they have to zip single dll first, would like be more  work for user in such case? How about we specify the another example usage of `mySparkApp args` as I mentioned earlier?

Reply:
If users are manually submitting then yes, but in "production" use, users would most likely automate the packaging and submission.  It would be good to know that they don't have to take different steps if they have single vs multiple dependencies.

Reply:
If we would like to unify these two sections, should we remove the command example in the first section?

I just feel like we need to add more contexts in this whole instruction in general, any suggestions would be really appreciated.